### PR TITLE
fix(events): Improve sorting of chapter events

### LINKED ISSUE #311: Event listing with upcoming and past separation
Currently events are getting displayed based on the feed we are getting, but there should be 2 sections:
1) Upcoming
2) Past

### PR DESCRIPTION
**Pull request by fnk0**
* Fixes the sorting of the events in EventListFragment. 

Sorry about the revert and recommit, etc.. the other commit was getting in the way! 
Having some branch problems, srry! 

Fixes #311

### CODE REVIEW

**Comment by tasomaniac** [comment]
Instead of only 1 commit, we will merge 5 commits. It will look ugly in the history. 
I can manually fix it and merge only one of them or can you fix it? :smile: 

You can create a new branch and cherry pick the right commit or you can rebase interactively and squash these into 1 commit.

File: app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java [hunk 1]
```diff
@@ -150,6 +155,44 @@ public void addEventToCalendar(SimpleEvent event) {
...
+        DateTime now = DateTime.now();
+
+        for (SimpleEvent event : events) {
+            if (event.getStart().isBefore(now)) {
+                past.add(event);
+            } else {
+                upcoming.add(event);
+            }
+        }
+        Collections.reverse(past);
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java]: I think we should reverse the upcoming list. Am I wrong? I am not sure.

File: app/src/main/java/org/gdg/frisbee/android/eventseries/TaggedEventSeriesFragment.java [hunk 2]
```diff
@@ -108,6 +108,7 @@ EventAdapter createEventAdapter() {
         return new EventAdapter(getActivity(), mTaggedEventSeries.getDefaultIconResId());
     }
 
+    @SuppressWarnings("unchecked")
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/eventseries/TaggedEventSeriesFragment.java]: These warnings are dangerous actually and we have to have something to fix that. These caused crashes in the past. 
Can you remove them? :smile:

File: app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java [hunk 3]
```diff
@@ -150,6 +155,44 @@ public void addEventToCalendar(SimpleEvent event) {
...
+
+    /**
+     * Split a events list into Upcoming / Past events
+     *
+     * @param events
+     *      The events list that needs to be sorted
+     * @return
+     *      A pair containting the split list into Upcoming / Past events
+     */
+    protected Pair<List<SimpleEvent>, List<SimpleEvent>> splitEventsList(Collection<? extends SimpleEvent> events) {
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java]: I think we can make this private.

**Comment by tasomaniac** [comment]
Thank you for the work. It looks great. 
I can also be done with a custom comparator where you compare things and decide which one to go first or not. 
You can look at `ChapterComparator` of ours.

**Comment by fnk0** [comment]
@tasomaniac I did a custom comparator at first but the problem is that there's a different behavior for each of them. That's why we reverse only the Past events list. If I used a custom comparator by date them we would have the same behavior even if the date was on the past. I might try later on making it all in a single Comparator and see if works. I only tried the comparator thing for about 10 minutes.

File: app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java (hunk 3 above)
Comment (fnk0) [inline app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java]: That is in the abstract fragment and I call it from 2 of the fragments that extends EventListFragment. That's why I made it protected.

File: app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java (hunk 1 above)
Comment (fnk0) [inline app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java]: No. This actually sorts like this. The Upcoming event list is sorted in ASC order as we want the ones that are the closest to the date we are first and the Past events order is sorted in DESC order as we want the most recents one on top.

File: app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java (hunk 3 above)
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java]: No I think you are calling the above in the other `splitEventsAndAddToAdapter ` Fragments. This one is not called.

Comment (fnk0) [inline app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java]: Got it! I can make it private them. And if in the future we just need to split the events without adding to the adapter we can change it to protected.

**Comment by tasomaniac** [comment]
1 comparator was just an idea. I think it can be done. You can control the behavior based on if the event is past or not. 
But I now think that it is not necessary because we have to split the Array into 2 anyways to display in 2 sections in the future.

File: app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java (hunk 1 above)
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/eventseries/EventListFragment.java]: You are absolutely right. I was confused. Thank you for the explanation.

**Comment by tasomaniac** [comment]
Rebased, deleted wrong commits, changed code review changes and manually merged.

**Pull request closed without merging**